### PR TITLE
fix: exclude e2e from CI coverage and remove dead ValueError guard (#472)

### LINKED ISSUE #472: [aw][code health] CI measures coverage over e2e tests; unreachable ValueError in _render_summary_header
## Root Cause

Two related code-health findings that together represent meaningful cleanup:

---

### 1. CI runs e2e tests inside coverage measurement — contradicts documented architecture

**File:** `.github/workflows/ci.yml`

The CI runs:
```yaml
- run: uv run pytest --cov --cov-report=xml -v
```

This collects coverage across **all** tests, including `tests/e2e/`. The architecture documentation explicitly states:

> *Coverage is measured on unit tests only (e2e coverage would be misleading)*
> — `src/copilot_usage/docs/architecture.md`

The `Makefile` correctly enforces this split:
```makefile
# Unit tests with coverage:
uv run pytest --ignore=tests/e2e $(TEST_FLAGS)
# E2e tests separately, without coverage:
uv run pytest tests/e2e -q --no-header
```

By including e2e tests in the coverage run, CI:
- Inflates overall coverage numbers (e2e tests exercise the same source paths as unit tests, making it easier to hit the coverage threshold with less unit-test discipline)
- Makes `diff-cover` results less meaningful (new lines might be "covered" only by e2e runs, not by proper unit tests)
- Contradicts the documented testing strategy

**Fix:** Update `ci.yml` to mirror the Makefile's split approach:
```yaml
- run: uv run pytest --ignore=tests/e2e --cov --cov-report=xml -v
- run: uv run pytest tests/e2e -v
- name: Check diff coverage (new/changed lines ≥ 90%)
  run: uv run diff-cover coverage.xml --compare-branch=origin/main --fail-under=90
```

---

### 2. Unreachable `ValueError` in `_render_summary_header`

**File:** `src/copilot_usage/report.py`

```python
def _render_summary_header(console, sessions):
    if not sessions:
        raise ValueError("_render_summary_header requires non-empty sessions")
    ...
```

Both call sites guard against this condition before calling the function:

```python
# render_summary (caller 1):
if not filtered:
    console.print("[yellow]No sessions found.[/yellow]")
    return
_render_summary_header(console, filtered)  # filtered is guaranteed non-empty

# render_full_summary (caller 2):
if not sessions:
    console.print("[yellow]No sessions found.[/yellow]")
    return
_render_summary_header(console, sessions)  # sessions is guaranteed non-empty
```

The `ValueError` is **dead code** — it can never be reached in practice. This creates a false impression that callers need to handle an exception from this helper, and it may show as an uncovered branch in coverage reports.

**Fix:** Remove the unreachable guard and document the precondition instead:

```python
def _render_summary_header(
    console: Console,
    sessions: list[SessionSummary],
) -> None:
    """Print the report header with date range.

    *sessions* must be non-empty — callers are responsible for the empty-list check.
    """
    # (remove the ValueError block entirely)
    start_times = [...]
```

---

### For finding 1 (CI coverage scope):
The fix is a CI configuration change. Verify by:
- Confirming `uv run pytest --ignore=tests/e2e --cov --cov-report=xml -v` still passes (it already does via `make test`)
- Confirming `uv run pytest tests/e2e -v` still passes independently

### For finding 2 (dead code):
Add a test that asserts `_render_summary_header` is never called with an empty list indirectly (it already cannot be reached), and remove any test that specifically tests the `ValueError` path (if one exists). After removing the guard, confirm `make check` passes cleanly — the dead branch is no longer a coverage gap.

> [!NOTE]
> <details>
> <summary>🔒 Integrity filtering filtered 13 items</summary>
>
> Integrity filtering activated and filtered the following items during workflow execution.
> This happens when a tool call accesses a resource that does not meet the required integrity or secrecy level of the workflow.
>
> - issue:microsasa/cli-tools#468 (`list_issues`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
> - issue:microsasa/cli-tools#467 (`list_issues`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
> - issue:microsasa/cli-tools#437 (`list_issues`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
> - issue:microsasa/cli-tools#221 (`list_issues`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
> - issue:microsasa/cli-tools#220 (`list_issues`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
> - issue:microsasa/cli-tools#173 (`list_issues`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
> - issue:microsasa/cli-tools#165 (`list_issues`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
> - issue:microsasa/cli-tools#131 (`list_issues`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
> - issue:microsasa/cli-tools#114 (`list_issues`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
> - issue:microsasa/cli-tools#111 (`list_issues`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
> - issue:microsasa/cli-tools#110 (`list_issues`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
> - issue:microsasa/cli-tools#103 (`list_issues`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
> - issue:microsasa/cli-tools#92 (`list_issues`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
>
> </details>

> Generated by [Code Health Analysis](https://github.com/microsasa/cli-tools/actions/runs/23692248315) · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsasa%2Fcli-tools%2Fcode-health%22&type=issues)

### LINKED ISSUE #477: fix: exclude e2e from CI coverage and remove dead ValueError guard (#472)
Closes #472

### 1. CI: exclude e2e tests from coverage measurement

Updated `.github/workflows/ci.yml` to mirror the Makefile's split approach:

- **Before:** `uv run pytest --cov --cov-report=xml -v` (runs all tests including e2e under coverage)
- **After:** Unit tests with coverage (`--ignore=tests/e2e`) + e2e tests separately (`tests/e2e -v`)

This aligns CI with the documented architecture (`architecture.md`: "Coverage is measured on unit tests only") and prevents e2e tests from inflating coverage numbers.

### 2. Remove unreachable `ValueError` in `_render_summary_header`

Both callers (`render_summary` and `render_full_summary`) already guard against empty sessions before calling `_render_summary_header`, making the `ValueError` dead code. Removed the guard and updated the docstring to document the precondition.

### 3. Test updates

- Removed `test_empty_sessions_raises_value_error` (tested the now-removed dead code path)
- Kept caller-guard tests confirming `render_summary` and `render_full_summary` handle empty lists correctly

## Verification

All 825 tests pass with 99.60% coverage (≥80% threshold met). Ruff, pyright, and format checks all clean.

---

> [!WARNING]
> 🛡️ **Protected Files — Push Permission Denied**
>
> This was originally intended as a pull request, but the patch modifies protected files: `.github/workflows/ci.yml`.
>
> The push was rejected because GitHub Actions does not have `workflows` permission to push these changes, and is never allowed to make such changes, or other authorization being used does not have this permission. A human must create the pull request manually.

To create a pull request with the changes:

```sh
# Download the patch from the workflow run
gh run download 23693610847 -n agent-artifacts -D /tmp/agent-artifacts-23693610847

# Create a new branch
git checkout -b fix/472-ci-coverage-and-dead-code-2d83dfb18b81f018 main

# Apply the patch (--3way handles cross-repo patches)
git am --3way /tmp/agent-artifacts-23693610847/aw-fix-472-ci-coverage-and-dead-code.patch

# Push the branch and create the pull request
git push origin fix/472-ci-coverage-and-dead-code-2d83dfb18b81f018
gh pr create --title 'fix: exclude e2e from CI coverage and remove dead ValueError guard (#472)' --base main --head fix/472-ci-coverage-and-dead-code-2d83dfb18b81f018 --repo microsasa/cli-tools
```

> [!WARNING]
> <details>
> <summary>⚠️ Firewall blocked 1 domain</summary>
>
> The following domain was blocked by the firewall during workflow execution:
>
> - `astral.sh`
>
> To allow these domains, add them to the `network.allowed` list in your workflow frontmatter:
>
> ```yaml
> network:
>   allowed:
>     - defaults
>     - "astral.sh"
> ```
>
> See [Network Configuration](https://github.github.com/gh-aw/reference/network/) for more information.
>
> </details>

> [!NOTE]
> <details>
> <summary>🔒 Integrity filtering filtered 1 item</summary>
>
> Integrity filtering activated and filtered the following item during workflow execution.
> This happens when a tool call accesses a resource that does not meet the required integrity or secrecy level of the workflow.
>
> - issue:microsasa/cli-tools#472 (`issue_read`: has lower integrity than agent requires. The agent cannot read data with integrity below "approved".)
>
> </details>

> Generated by [Issue Implementer](https://github.com/microsasa/cli-tools/actions/runs/23693610847) · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+%22gh-aw-workflow-id%3A+issue-implementer%22&type=pullrequests)

### PR DESCRIPTION
**Pull request by microsasa**
Applies the patch from the issue-implementer agent (workflow run #23693610847). The agent couldn't push because `ci.yml` is a protected file.

### Changes
1. **CI**: split test run — unit tests with coverage (`--ignore=tests/e2e`), e2e tests separately. Aligns CI with Makefile and documented architecture.
2. **report.py**: removed unreachable `ValueError` guard in `_render_summary_header` (both callers already check for empty sessions).
3. **Tests**: removed test for the now-removed dead code path.

All 825 tests pass, 99% coverage, lint/typecheck clean.

Closes #472
Closes #477

### CODE REVIEW

File: tests/copilot_usage/test_report.py [hunk 1]
```diff
@@ -4844,28 +4844,19 @@ def test_max_events_one_shows_last(self) -> None:
 
 
 # ---------------------------------------------------------------------------
-# Issue #446 — Cleanup 1: _render_summary_header ValueError guard
+# Issue #446 — Cleanup: callers guard against empty sessions
```
[the remark this answers is not in the document]
Comment (microsasa) [reply tests/copilot_usage/test_report.py]: Fixed — updated to #472.
